cmplibrary/src/main/java/com/sourcepoint/cmplibrary/consent/ClientEventManager.kt

```diff
@@ -34,42 +31,35 @@ private class ClientEventManagerImpl(
     val consentManagerUtils: ConsentManagerUtils,
 ) : ClientEventManager {
 
-    private var cNumber: Int = Int.MAX_VALUE
-    private var storedConsent: Int = 0
-
-    override fun setCampaignNumber(campNum: Int) {
-        cNumber = campNum
-        storedConsent = 0
-    }
+    private var campaignsToProcess: Int = Int.MAX_VALUE
+    private var isFirstLayerMessage = true
 
-    override fun executingLoadPM() {
-        cNumber = 1
-        storedConsent = 0
+    override fun setCampaignsToProcess(numberOfCampaigns: Int) {
+        campaignsToProcess = numberOfCampaigns
     }
 
     override fun setAction(action: ConsentActionImpl) {
         executor.executeOnSingleThread {
             when (action.actionType) {
-                ActionType.ACCEPT_ALL,
-                ActionType.REJECT_ALL,
-                ActionType.SAVE_AND_EXIT -> {
-                }
                 ActionType.SHOW_OPTIONS -> {
-                }
-                ActionType.UNKNOWN -> {
+                    isFirstLayerMessage = false
                 }
                 ActionType.CUSTOM,
                 ActionType.MSG_CANCEL,
                 ActionType.PM_DISMISS -> {
-                    if (!action.requestFromPm || action.singleShotPM) {
-                        if (cNumber > 0) cNumber--
+                    if (isFirstLayerMessage) {
+                        campaignsToProcess--
                     }
+                    isFirstLayerMessage = true
```

Comment:
why if we dismiss the PM we set `isFirstLayerMessage ` as `true` in this case?

Reply:
There are 3 cases in which a user find themselves in the PM.
1. the user landed in the Privacy Manager via the 1st layer message. In this case, dismissing the PM should should return the user to the first layer message, thus we should reset the value of `isFirstLayerMessage`
2. the PM _is_ the first layer message, in this case, dismissing the PM should close the message entirely, here we reset `isFirstLayerMessage` just to have that variable at its "original" state of `true`.
3. the PM is loaded on demand (ie. via `loadPrivacyManager` methods). Same as the item before, simply reseting the value to `true` (even though its value should be `true` anyways).